addons/mail/models/mail_thread.py

```diff
@@ -1331,7 +1332,7 @@ def _message_parse_extract_bounce(self, email_message, message_dict):
         if not isinstance(email_message, Message):
             raise TypeError('message must be an email.message.Message at this point')
 
-        email_part = next((part for part in email_message.walk() if part.get_content_type() == 'message/rfc822'), None)
+        email_part = next((part for part in email_message.walk() if part.get_content_type() in {'message/rfc822', 'text/rfc822-headers'}), None)
```

Comment:
Customer is asking for adding support for ``message/rfc822-headers`` content-type, which is not standard, but was in draft specification https://tools.ietf.org/id/draft-ietf-notary-mime-report-01.txt

Reply:
i.e.

        email_part = next((part for part in email_message.walk() if part.get_content_type() in {'message/rfc822', 'message/rfc822-headers', 'text/rfc822-headers'}), None)

Reply:
Because in their samples, it's
``message/rfc822-headers``
and not 
``text/rfc822-headers``

opw-2344252